edx_exams/apps/lti/views.py

```diff
@@ -6,338 +6,91 @@
-    # opaque to the Proctoring Tool in the Start Proctoring message.
-    # This will be returned in the Start Assessment message and acts as an anti-CSRF token,
-    # the Assessment Tool MUST verify that this data matches the expected browser session
-    # before actually starting the assessment."
-    # See 3.3 Transferring the Candidate Back to the Assessment Platform.
-    # In the synchronizer token method of CSRF protection, the anti-CSRF token must be stored somehow on the server.
-    # Note that this generates an anti-CSRF token PER USER SESSION, not per request. Per request is more secure, but
-    # I elected to do is on the session for the POC, because it is simpler.
-    session_data = request.session.get('lti_proctoring_session_data')
-    if session_data is None:
-        session_data = get_random_string(32)
-        request.session['lti_proctoring_session_data'] = session_data
+    exam_attempt = ExamAttempt.objects.get(pk=attempt_id)
+    exam = exam_attempt.exam
+    lti_config_id = exam.provider.lti_configuration_id
+    lti_config = LtiConfiguration.objects.get(id=lti_config_id)
 
-    start_assessment_url = urljoin(settings.ROOT_URL, reverse('lti:start-assessment'))
-
-    # TODO: The resource link should uniquely represent the assessment in the Assessment Platform.
-    # TODO: We SHOULD provide a value for the title attribute.
-    # TODO: It's RECOMMENDED to provide a value for the description attribute.
-    # TODO: The xblock-lti-consumer library does not currently support setting these attributes.
-    resource_link = get_resource_link()
-
-    lti_consumer.enable_proctoring(
-        # NOTE TO SELF: attempt_number is an auto-incrementing integer from 1 per learner, per assessment.
-        29,  # attempt_number
-        session_data,
-        resource_link,
-        start_assessment_url=start_assessment_url,
+    proctoring_start_assessment_url = urljoin(
+        settings.ROOT_URL,
+        reverse('lti_consumer:lti_consumer.start_proctoring_assessment_endpoint')
     )
 
-    # This is necessary for testing with the IMS tool, since the user will be an AnonymousUser without an id.
-    # TODO: Replace this with the authenticated user's id.
-    # TODO: Remove this once testing is complete.
-    user_id = 1 if request.user.id is None else request.user.id
-
-    # Required user claim data
-    lti_consumer.set_user_data(
-        #   user_id=request.user.id,
-        user_id=user_id,
-        # Pass Django user role to library
-        role='student'
+    proctoring_launch_data = Lti1p3ProctoringLaunchData(
+        # TODO: This is incorrect. It should be an integer starting at 1 and increasing by 1 for each subsequent attempt
+        #       per user.
+        attempt_number=exam_attempt.id,
+        start_assessment_url=proctoring_start_assessment_url,
     )
 
-    # These claims are optional.
-    # TODO: This will need to have additional consideration for PII.
```

Comment:
Question: I'm guessing that this todo is addressed below?

Reply:
We aren't sending over any PII for proctoring in this implementation anymore, so I have removed the TODO. We can add it back in after things are ironed out with product and legal.